pkg/api/resource/status.go

```diff
@@ -109,6 +109,8 @@ const (
 	StatusAPIGroupParentFailed
 	StatusAPIGroupParentKilled
 	StatusAPIGroupUpdateSkipped
+
+	StatusDataOOM
```

Comment:
Should we add StatusDataParentKilledOOM and StatusAPIGroupParentKilledOOM? Another option is to reuse StatusDataParentKilled and StatusAPIGroupParentKilled if you don’t think it’s necessary to differentiate these two on the downstream resources.